packages/atomic/src/components/common/checkbox.spec.ts

```diff
@@ -0,0 +1,125 @@
+import {fireEvent, within} from '@storybook/test';
+import {html, render} from 'lit';
+import {vi} from 'vitest';
+import {checkbox, CheckboxProps} from './checkbox';
+
+describe('checkbox', () => {
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  const renderCheckbox = (props: Partial<CheckboxProps>): HTMLButtonElement => {
+    render(
+      html`${checkbox({
+        ...props,
+        checked: props.checked ?? false,
+        onToggle: props.onToggle ?? vi.fn(),
+      })}`,
+      container
+    );
+    return within(container).getByRole('checkbox') as HTMLButtonElement;
+  };
+
+  it('should render a checkbox with the correct text attributes', () => {
+    const props = {
+      text: 'Test Checkbox',
+      ariaLabel: 'Test Checkbox',
+    };
+
+    const button = renderCheckbox(props);
+
+    expect(button).toBeInTheDocument();
+    expect(button.getAttribute('aria-label')).toBe('Test Checkbox');
+    expect(button.value).toBe('Test Checkbox');
+  });
+
+  it('should not be checked by default', async () => {
+    const button = renderCheckbox({});
+
+    expect(button.getAttribute('aria-checked')).toBe('false');
+    expect(button.classList.contains('selected')).toBe(false);
+  });
+
+  it('should be have selected attributes and classes if checked', async () => {
+    const button = renderCheckbox({checked: true});
+
+    expect(button.getAttribute('aria-checked')).toBe('true');
+    expect(button.classList.contains('selected')).toBe(true);
+  });
+
+  it('should call onToggle when the checkbox is clicked', async () => {
+    const onToggle = vi.fn();
+    const props = {
+      onToggle,
+    };
+
+    const button = renderCheckbox(props);
+
+    await fireEvent.click(button);
+
+    expect(onToggle).toHaveBeenCalledWith(true);
+  });
+
+  it('should render a checkbox with the correct class', () => {
+    const props = {
+      class: 'test-class',
+    };
+
+    const button = renderCheckbox(props);
+
+    expect(button).toBeInTheDocument();
+    expect(button.classList.contains('test-class')).toBe(true);
+  });
+
+  it('should render a checkbox with the correct part attribute', () => {
```

Comment:
I think we should also test the default value when no `part` prop is specified (`checkbox`), as well as validate that the `iconPart` prop works.

Reply:
Also, we should validate that a new part is added when the `checked` prop is `true`.

Reply:
good point